test/functional/p2p_compactblocks.py

```diff
@@ -258,22 +251,14 @@ def check_announcement_of_new_block(node, peer, predicate):
         test_node.send_and_ping(msg_sendheaders())
         check_announcement_of_new_block(node, test_node, lambda p: "cmpctblock" in p.last_message)
 
-        # Try one more time, after sending a version-1, announce=false message.
-        test_node.send_and_ping(msg_sendcmpct(announce=False, version=preferred_version-1))
+        # Try one more time, after sending a version=1, announce=false message.
+        test_node.send_and_ping(msg_sendcmpct(announce=False, version=1))
         check_announcement_of_new_block(node, test_node, lambda p: "cmpctblock" in p.last_message)
 
         # Now turn off announcements
-        test_node.send_and_ping(msg_sendcmpct(announce=False, version=preferred_version))
+        test_node.send_and_ping(msg_sendcmpct(announce=False, version=2))
         check_announcement_of_new_block(node, test_node, lambda p: "cmpctblock" not in p.last_message and "headers" in p.last_message)
 
-        if old_node is not None:
```

Comment:
If I'm reading correctly, I think we want to keep this test but reverse the expected outcome? That way we test that v1 doesn't work anymore as opposed to assuming it doesn't.

Disregard if I'm missing something else that exercises this.

Reply:
No, that's a good point. I've added the test in commit _[net processing] Only accept SENDCMPCT with version = 2_